proposal/MazenSli/proposal.rst

```diff
@@ -0,0 +1,76 @@
+
+Image registration is the process of aligning two or more images of the same scene, i.e. to transform them into the same coordinate system. It is necessary in order to be able to compare or/and to retrieve data from these different measurements. As for medical data for instance, it can be patient measurements from different times or measurement data from different patients (monomodal) or measurement data from different modalities (e.g. CT, MRT, PET) of one patient (multimodal). The process consists of initializing one image as the reference image, also called the fixed image, and applying geometric transformations or local displacements to the other images, also called moving images, to align them with the reference.
+
+From a mathematical point of view, this means that the transformation can be for example rigid, i.e. consisting of translations and rotations or affine (e.g. shears). But for more complex applications, for example organs and tissue that are moving or changing their form or also for patient-to-patient registration, a rigid transformation is inappropriate. In this cases we need a deformable (nonlinear) registration. With an increase of degrees of freedom in the transformation we get closer to the model of the real physical deformation. Examples non-rigid registration are: Spline functions (e.g. B-Splines), elastic registration, fluid registration and Finite Element Models (FEM).
+
+Can we use image registration for data augmentation?
+
+Of course! Let say there is for example we have a limited set of brain-tumor training sets. To extend our data set, we can use affine transformations to produce correlated images and anatomically incorrect examples, or use elastic transformations which can bring noise into the data.
+
+Perspective users
+=================
+
+The library Rising has not yet any functions for image regsitration. The task would be a robust implementation of the mentioned algorithms and subsequently the corresponding unittesting. The library consists of the two moduls loading and transforms. The project would be a new sub-modul rising.transforms.registration including all the implemented functions.
+
+System architecture and API description
+=======================================
+
+The library Rising can load 2D and 3D data and provides several transform algorithms for for data augmentation purposes. For more information about the API see: https://github.com/PhoenixDL/rising
+
+The sub-modul rising.transforms.registration will consist of several image registration functions (e.g. elastic registration) which take a reference image and the moving images (one or more), aligns the moving images according to the reference and outputs the aligned moving images. It will also include the corresponding unittesting.
+
+Engineering infrastructure
+==========================
+
+The project will make use of PyTorch and several other Python libraries such as SimpleITK, natari, numpy and matplotlib.
```

Comment:
Is there a role for C++ to play in your project?

The upstream project (rising) seems to be purely in Python.  How do you plan to incorporate the build system of C++ in it?